src/dash-table/components/EdgeFactory.tsx

```diff
@@ -0,0 +1,228 @@
+import derivedHeaderEdges from 'dash-table/derived/edges/header';
+import derivedHeaderOpEdges from 'dash-table/derived/edges/operationOfHeaders';
+import { EdgesMatrices, IEdgesMatrices } from 'dash-table/derived/edges/type';
+
+import getHeaderRows from 'dash-table/derived/header/headerRows';
+
+import { derivedRelevantCellStyles, derivedRelevantFilterStyles, derivedRelevantHeaderStyles } from 'dash-table/derived/style';
+import { Style, Cells, DataCells, BasicFilters, Headers } from 'dash-table/derived/style/props';
+
+import { ControlledTableProps, VisibleColumns, IViewportOffset, Data } from './Table/props';
+
+export default class EdgeFactory {
+    private readonly dataStyles = derivedRelevantCellStyles();
+    private readonly dataOpStyles = derivedRelevantCellStyles();
+    private readonly filterStyles = derivedRelevantFilterStyles();
+    private readonly filterOpStyles = derivedRelevantFilterStyles();
+    private readonly headerStyles = derivedRelevantHeaderStyles();
+    private readonly headerOpStyles = derivedRelevantHeaderStyles();
+
+    private readonly getDataEdges = derivedDataEdges();
+    private readonly getDataOpEdges = derivedDataOpEdges();
+    private readonly getFilterEdges = derivedFilterEdges();
+    private readonly getFilterOpEdges = derivedFilterOpEdges();
+    private readonly getHeaderEdges = derivedHeaderEdges();
+    private readonly getHeaderOpEdges = derivedHeaderOpEdges();
+
+    private hReconcile(target: EdgesMatrices | undefined, next: EdgesMatrices | undefined, cutoffWeight: number): EdgesMatrices | undefined {
+        if (!target || !next) {
+            return target;
+        }
+
+        target = target.clone();
+
+        const hNext = next.getMatrices().horizontal;
+        const hTarget = target.getMatrices().horizontal;
+
+        const iNext = 0;
+        const iTarget = hTarget.rows - 1;
+
+        R.forEach(j =>
+            (
+                hNext.getWeight(iNext, j) > cutoffWeight ||
+                (hNext.getWeight(iNext, j) > hTarget.getWeight(iTarget, j))
+            ) && hTarget.setEdge(iTarget, j, undefined, -Infinity, true),
+            R.range(0, hTarget.columns)
+        );
+
+        return target;
+    }
+
+    private vReconcile(target: EdgesMatrices | undefined, next: EdgesMatrices | undefined, cutoffWeight: number): EdgesMatrices | undefined {
+        if (!target || !next) {
+            return target;
+        }
+
+        target = target.clone();
+
+        const vNext = target.getMatrices().vertical;
+        const vTarget = target.getMatrices().vertical;
+
+        const jNext = 0;
+        const jTarget = vTarget.columns - 1;
+
+        R.forEach(i =>
+            (
+                vNext.getWeight(i, jNext) > cutoffWeight ||
+                (vNext.getWeight(i, jNext) > vTarget.getWeight(i, jTarget))
+            ) && vTarget.setEdge(i, jTarget, undefined, -Infinity, true),
+            R.range(0, vTarget.rows)
+        );
+
+        return target;
+    }
```

Comment:
Evaluates boundary vertical edges between the target edges matrix and the "next" edges matrix.